presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleStreamReader.java

```diff
@@ -91,37 +103,68 @@ public Block readBlock(Type type)
             }
             if (readOffset > 0) {
                 if (dataStream == null) {
-                    throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but data stream is not present");
+                    throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but data stream is missing");
                 }
                 dataStream.skip(readOffset);
             }
         }
 
-        BlockBuilder builder = type.createBlockBuilder(null, nextBatchSize);
-        if (presentStream == null) {
-            if (dataStream == null) {
-                throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but data stream is not present");
+        Block block;
+        if (dataStream == null) {
+            if (presentStream == null) {
+                throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is null but present stream is missing");
             }
-            dataStream.nextVector(type, nextBatchSize, builder);
+            presentStream.skip(nextBatchSize);
+            block = RunLengthEncodedBlock.create(DOUBLE, null, nextBatchSize);
+        }
+        else if (presentStream == null) {
+            block = readNonNullBlock();
         }
         else {
-            for (int i = 0; i < nextBatchSize; i++) {
-                if (presentStream.nextBit()) {
-                    verify(dataStream != null);
-                    type.writeDouble(builder, dataStream.next());
-                }
-                else {
-                    builder.appendNull();
-                }
+            boolean[] isNull = new boolean[nextBatchSize];
+            int nullCount = presentStream.getUnsetBits(nextBatchSize, isNull);
+            if (nullCount == 0) {
+                block = readNonNullBlock();
+            }
+            else if (nullCount != nextBatchSize) {
+                block = readNullBlock(isNull, nextBatchSize - nullCount);
+            }
+            else {
+                block = readNullBlock(isNull, nextBatchSize - nullCount);
```

Comment:
This line is for the all null case, but it's exactly the same as the case of (nullCount != nextBatchSize). Why is there this else{} block? Should it still be 
`block = RunLengthEncodedBlock.create(type, null, nextBatchSize);`

Reply:
I think this might be an issue in prestosql as well. Nice find @yingsu00 !

Reply:
It's not an issue in prestosql: https://github.com/prestosql/presto/blob/master/presto-orc/src/main/java/io/prestosql/orc/reader/DoubleStreamReader.java#L135